docs/csharp/fundamentals/program-structure/main-command-line.md

```diff
@@ -105,64 +108,81 @@ Execution succeeded
 Return value = 0
 ```
 
-### Async Main return values
+### Main return Task
+
+When the application entry point returns a `Task` or `Task<int>`, the compiler generates a new entry point that calls the entry point method declared in the application code. Assuming that this entry point is called `$GeneratedMain`, the compiler generates the following code for these entry points:
+
+- `static Task Main()` results in the compiler emitting the equivalent of `private static void $GeneratedMain() => Main().GetAwaiter().GetResult();`
+- `static Task Main(string[])` results in the compiler emitting the equivalent of `private static void $GeneratedMain(string[] args) => Main(args).GetAwaiter().GetResult();`
+- `static Task<int> Main()` results in the compiler emitting the equivalent of `private static int $GeneratedMain() => Main().GetAwaiter().GetResult();`
+- `static Task<int> Main(string[])` results in the compiler emitting the equivalent of `private static int $GeneratedMain(string[] args) => Main(args).GetAwaiter().GetResult();`
+
+> [!NOTE]
+>If `Main` include the `async` modifier, the compiler would generate the same code.
```

Comment:
I'm not sure what you're trying to say here. Is it that adding the `async` modifier causes the specific code gen, whether or not the `Main` method has an `await` expression?

Reply:
I removed this part about boilerplate.